### PR TITLE
Submit app.resp.RESP as replacement for dev.rdm.RDM

### PR DESCRIPTION
**Pull request by uglide**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. https://github.com/uglide/RedisDesktopManager
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by uglide** [comment]
bot, build app.resp.RESP

**Comment by hfiguiere** [comment]
what do you mean as a replacement? Does the other one need to be marked as EOL?

**Comment by barthalion** [comment]
Yes, turns out RDM is trademarked in the USA. Long story.

File: app.resp.RESP.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "resp",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=~/.rdm:create",
```
Comment (hfiguiere) [inline app.resp.RESP.json:13]: shouldn't that be `--persist` instead?

File: app.resp.RESP.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            "builddir": true,
+            "build-commands": [
+                "pushd ./3rdparty/lz4/build/cmake && cmake -DBUILD_STATIC_LIBS=true && make -j $FLATPAK_BUILDER_N_JOBS && popd",
+                "pushd ./3rdparty/zstd/build/cmake && cmake -DZSTD_BUILD_PROGRAMS=OFF . && make -j $FLATPAK_BUILDER_N_JOBS && popd",
+                "pushd ./3rdparty/snappy && cmake -DSNAPPY_BUILD_BENCHMARKS=OFF -DSNAPPY_BUILD_TESTS=OFF . && make -j $FLATPAK_BUILDER_N_JOBS && popd",
+                "pushd ./3rdparty/brotli && cmake . && make -j $FLATPAK_BUILDER_N_JOBS && ls -ahl && popd",
+                "pushd ./src && lrelease resp.pro && qmake LIBS+=-L/app/lib VERSION=\"$(git describe --abbrev=0 --tags|echo '2022')+$(git rev-parse --short HEAD)\" && make -j $FLATPAK_BUILDER_N_JOBS && popd",
+                "cp ./bin/linux/release/resp $FLATPAK_DEST/bin"
+            ],
+            "post-install": [
```
Comment (hfiguiere) [inline app.resp.RESP.json:47]:
```suggestion
```

You don't need a post-install phase in that case.

File: app.resp.RESP.json (hunk 1 above)
Comment (barthalion) [inline app.resp.RESP.json:13]: Just to double-check, this is because RESP still uses `~/.rdm` for configuration files?

**Comment by uglide** [comment]
@hfiguiere old package was marked as EOL already https://github.com/flathub/dev.rdm.RDM/blob/master/flathub.json

File: app.resp.RESP.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{
+    "app-id": "app.resp.RESP",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
```
Comment (barthalion) [inline app.resp.RESP.json:4]: A newer runtime `5.15-21.08` is out, we can give it a shot while we're on it.

File: app.resp.RESP.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "resp",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
```
Comment (barthalion) [inline app.resp.RESP.json:9]:
```suggestion
```

This is not needed when `fallback-x11` is passed.

File: pip-dependencies.json [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            "name": "python3-phpserialize",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \".\" --no-build-isolation"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/mrnom/phpserialize.git",
+                    "commit": null
```
Comment (hfiguiere) [inline pip-dependencies.json:16]: you need a commit here or a tag

**Review by barthalion — APPROVED** [review]

**Comment by barthalion** [comment]
bot, build app.resp.RESP

File: app.resp.RESP.json [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                    "url": "https://github.com/uglide/RedisDesktopManager",
+                    "commit": "a2c7bec957fe13d351c88e5ec682d672ff7275a0",
+                    "x-checker-data": {
+                        "type": "git",
+                        "tag-pattern": "^([\\d.]+)$"
+                    }
+                },
+                {
+                    "type": "archive",
+                    "url": "https://rdm-dev.nyc3.digitaloceanspaces.com/snap/deps.zip?v3",
```
Comment (hfiguiere) [inline app.resp.RESP.json:64]: is that file stable? ie will it ever change at that URL?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: app.resp.RESP.json (hunk 1 above)
Comment (uglide) [reply app.resp.RESP.json:13]: Yes, for backward compatibility.

**Comment by uglide** [comment]
bot, build app.resp.RESP

**Comment by barthalion** [comment]
bot, build app.resp.RESP

**Comment by barthalion** [comment]
bot, build app.resp.RESP

**Comment by uglide** [comment]
@barthalion Thanks!

**Pull request closed without merging**